### PR TITLE
docs/refine code doc

### PR DESCRIPTION
**Pull request by Noggling**
# Description

Adding code docs and some more documentation.
- Observable
- Mediator
- Moved files
- Delete old tech spec workspace core.

Completes: [AB#31679](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/31679)
[AB#30474](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/30474)

## Type of change

Remove options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read.
-   [x] Documentation and comments is added where needed.
-   [ ] My code is covered by tests.
-   [ ] UX has reviewed relevant UI contributions.

### CODE REVIEW

File: documentation/src/packages/workspace-core/mediator.md [hunk 1]
```diff
@@ -8,4 +8,127 @@ tags:
...
+    end
+
+    wsDataSourceConfig <--> wc
+    wsFilterConfig <--> wc
+
+```
+
+## Middleware
+
+To allow for complete flexibility an `AddMiddleware` function is added. This allows access to th e mediator. This should be used with caution.
```
Comment (Gustav-Eikaas) [inline documentation/src/packages/workspace-core/mediator.md:65]: the

File: documentation/src/packages/workspace-core/mediator.md [hunk 2]
```diff
@@ -8,4 +8,127 @@ tags:
...
+The example below is a simple data logger connecting to the onDataChange event.
+
+```TS
+    mediator.AddMiddleware((mediator) => {
+      mediator.onDataChange((data)=> {
```
Comment (Gustav-Eikaas) [inline documentation/src/packages/workspace-core/mediator.md:70]: mediator.dataService now, not so important tho

**Review by Gustav-Eikaas — APPROVED** [review]
LGTM

File: documentation/src/packages/workspace-core/mediator.md (hunk 2 above)
Comment (Noggling) [reply documentation/src/packages/workspace-core/mediator.md:70]: will fix :)

**Pull request merged**